src/container/FluxContainer.js

```diff
@@ -153,7 +153,10 @@ function create<DefaultProps, Props, State>(
         super.componentWillReceiveProps(nextProps, nextContext);
       }
 
-      if (realOptions.withProps || realOptions.withContext) {
+      if (
+        realOptions.withProps && !shallowEqual(this.props, nextProps) ||
```

Comment:
You don't have any sort of guarantee that the props are pure, that's why performance takes a big hit when you use `withProps` and why we recommend against it in the docs